### PR TITLE
Pivotal ID #185788503: Reactive Mongo

### PR DESCRIPTION
**Pull request by Juan-EBI**
https://www.pivotaltracker.com/story/show/185788503

### CODE REVIEW

File: commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/integration/SerializationService.kt [hunk 1]
```diff
@@ -15,13 +16,15 @@ interface SerializationService {
 
     fun serializeFileList(files: Sequence<BioFile>, targetFormat: SubFormat, outputStream: OutputStream)
```
Comment (jhoanmanuelms) [inline commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/integration/SerializationService.kt:17]: Do we still need this method?

File: submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/mapping/to/ToExtFileListMapperTest.kt [hunk 2]
```diff
@@ -59,14 +61,14 @@ class ToExtFileListMapperTest(temporaryFolder: TemporaryFolder) {
...
-    fun `toExtFileList including FileListFiles`() {
+    fun `toExtFileList including FileListFiles`() = runTest {
         every {
             fileListDocFileRepository.findAllBySubmissionAccNoAndSubmissionVersionAndFileListName(
                 "S-TEST123",
                 1,
                 "file-list"
             )
-        } returns listOf(fileListDocFile)
+        } returns Flux.fromIterable(listOf(fileListDocFile))
```
Comment (jhoanmanuelms) [inline submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/mapping/to/ToExtFileListMapperTest.kt:71]: image

**Review by jhoanmanuelms — APPROVED** [review]
Great changes man. LGTM

File: commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/integration/SerializationService.kt (hunk 1 above)
Comment (Juan-EBI) [reply commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/integration/SerializationService.kt:17]: we actually need this for example when we serialize a table (limited number of files)

**Pull request merged**